pkg/protocol/xprotocol/dubbo/mapping.go

```diff
@@ -0,0 +1,37 @@
+package dubbo
+
+import (
+	"context"
+	"errors"
+	"mosn.io/mosn/pkg/protocol/xprotocol"
+	"mosn.io/mosn/pkg/types"
+	"net/http"
+)
+
+func init() {
+	xprotocol.RegisterMapping(ProtocolName, &dubboStatusMapping{})
+}
+
+type dubboStatusMapping struct{}
+
+func (m *dubboStatusMapping) MappingHeaderStatusCode(ctx context.Context, headers types.HeaderMap) (int, error) {
+	cmd, ok := headers.(xprotocol.XRespFrame)
+	if !ok {
+		return 0, errors.New("no response status in headers")
+	}
+	code := uint16(cmd.GetStatusCode())
+	switch code {
+	case 20:
+		return http.StatusOK, nil
+	case 30, 31:
+		return http.StatusGatewayTimeout, nil
```

Comment:
dubbo framework status code definition:  https://github.com/apache/dubbo/blob/master/dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/Response.java

    /**
     * clien side timeout.
     */
    public static final byte CLIENT_TIMEOUT = 30;

    /**
     * server side timeout.
     */
    public static final byte SERVER_TIMEOUT = 31;


You must ensure that all status codes do not break the original semantics